src/backend/types/metadata-internal.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+export module v1Types {
```

Comment:
I think it would be better to move metadata related types to one file (metadata.ts) and also use versions in the types names:
```ts
export interface MetadataFileV1 {}

export interface MetadataFileV2 {}
```

Reply:
I don't think the non-v2 types will be used outside the parser, so calling the one exported type MetadataFileVxyz just adds clutter.

I could move it to a single file, but in a separate internal file it won't be available via ``import ... from 'backend/types'``,
only from ``import ... from 'backend/types/metadata-internal'``
(or under ``'backend/types/internal/metadata'``, as that is a better layout).